transmit/direct_transmit.go

```diff
@@ -359,11 +378,10 @@ func (d *DirectTransmission) sendBatch(wholeBatch []*types.Event) {
 func (d *DirectTransmission) batchEvents(in <-chan *types.Event) {
 	defer d.shutdownWG.Done()
 
-	var batch []*types.Event
 	for ev := range in {
 		// Start a new batch, and wait for the first event.
-		batch = batch[:0]
-		batch = append(batch, ev)
+		batch := make([]*types.Event, 1, d.MaxBatchSize)
```

Comment:
Is the reason we can’t reuse the underlying array because each worker needs its own independent copy of the batched events?

Reply:
Because we hand it to sendBatch(), which now runs concurrently. So we can't overwrite the contents of the slice, because that would mangle the data we gave to sendBatch(). Now, the slice is created, filled, passed to sendBatch(), and then becomes garbage.

A more efficient approach would recycle these via a pool, and I may do that at some point.